pytential/linalg/proxy.py

```diff
@@ -330,9 +329,9 @@ def __call__(self,
 
         # {{{ get proxy centers and radii
 
-        from arraycontext import thaw
-        from meshmode.dof_array import flatten
-        sources = flatten(thaw(discr.nodes(), actx))
+        sources = actx.np.reshape(
+                flatten(discr.nodes(), actx),
+                (self.ambient_dim, -1))
```

Comment:
This is not an improvement IMO. The prior approach involved no copying, whereas this forces a copy. Do we have a way in `arraycontext` to flatten only below a certain `leaf_class`? If not, then I think this would be a good use case for it.

Reply:
No, it just flattens the whole thing at the moment.

Is it worth adding there or should we just un-deprecate `meshmode.dof_array.flatten`? And implement it in terms of `arraycontext.flatten` mostly.

Reply:
I think it makes sense to add to `arraycontext.flatten`. It might be relatively simple to do.

Reply:
Assuming inducer/arraycontext#142 gets in, cce2995 removes all uses of `reshape + flatten` like that.

Reply:
f9d5fcd should fix any remaining issues.